### PR TITLE
Add --snapshot-property argument to pass to zfs snapshot

### PR DESCRIPTION
**Pull request by digitalsignalperson**
Just exposing the property argument of zfs snapshot

From zfs snapshot man page:

```
     zfs snapshot [-r] [-o property=value]… dataset@snapname…

     -o property=value
         Set the specified property; see zfs create for details.
```

This is a different effect than changing a property on received. Also not the same as a pre-snapshot `zfs set`. A possible use-case is tracking info that should live with the atomic snapshot.

Example of pulling snapshots from different nodes and tracking where the snapshot came from
```
NAME                                              USED  WRITTEN:BY
rpool/DATA/project@data-2022-06-12_22-44-42        64K  node1
rpool/DATA/project@data-2022-06-12_22-45-41        64K  node2
rpool/DATA/project@data-2022-06-12_22-47-02        72K  node1
rpool/DATA/project@data-2022-06-12_22-48-17         0B  node3
```

### CODE REVIEW

**Comment by digitalsignalperson** [comment]
Also a bit curious on zfs properties behaviour. I couldn't find the explanation in the zfs docs, but find that properties are NOT creating a persistent history with `zfs set`, but they do when using `zfs snapshot -o property=value`

e.g. no history of property changes like this

```
zfs create rpool/zzz
zfs snapshot rpool/zzz@1
zfs set test:test=1 rpool/zzz
zfs snapshot rpool/zzz@2
zfs set test:test=2 rpool/zzz
zfs snapshot rpool/zzz@3
zfs set test:test=3 rpool/zzz

zfs list -t snapshot -o name,test:test rpool/zzz
NAME         TEST:TEST
rpool/zzz@1  3
rpool/zzz@2  3
rpool/zzz@3  3
```

vs

```
zfs create rpool/yyy
zfs snapshot -o test:test=1 rpool/yyy@1
zfs snapshot -o test:test=2 rpool/yyy@2
zfs snapshot -o test:test=3 rpool/yyy@3

zfs list -t snapshot -o name,test:test rpool/yyy
NAME         TEST:TEST
rpool/yyy@1  1
rpool/yyy@2  2
rpool/yyy@3  3
```

is that a zfs bug or intended behaviour?

**Comment by psy0rz** [comment]
can you call it --set-snapshot-properties and make it behave like --set-properties? (e.g. commaseperated multiple properties)

**Comment by psy0rz** [comment]
very interesting. I can imagine they did this so that the properties on a snapshot are "frozen" as well, just like the snapshot itself? e.g. you can only set them when creating the snapshot, and never change them again after that?

**Comment by psy0rz** [comment]
no i meant user properties that are set during zfs snapshot -o test:test=blah

it would make sense that they are immutable after creating the snapshot

File: zfs_autobackup/ZfsNode.py [hunk 1]
```diff
@@ -212,6 +212,8 @@ def consistent_snapshot(self, datasets, snapshot_name, min_changed_bytes, pre_sn
             # create consistent snapshot per pool
             for (pool_name, snapshots) in pools.items():
                 cmd = ["zfs", "snapshot"]
+                for snapshot_property in set_snapshot_properties
```
Comment (digitalsignalperson) [inline zfs_autobackup/ZfsNode.py:215]: missing ':'

**Comment by digitalsignalperson** [comment]
Hmm aside from syntax error in that last change, the patch is no longer working for me...

My original testing was branched off v3.1.1, not sure where I messed up yet

**Comment by Scrin** [comment]
Correct, user properties on snapshots are mutable as well. You could for example use user properties on snapshots to mark which of the snapshots have been already synchronized elsewhere

**Comment by psy0rz** [comment]
thanks for your contribution :)

**Pull request merged**